cocos/misc/lodgroup-component.ts

```diff
@@ -0,0 +1,479 @@
+
+ https://www.cocos.com/
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated engine source code (the "Software"), a limited,
+ worldwide, royalty-free, non-assignable, revocable and non-exclusive license
+ to use Cocos Creator solely to develop games on your target platforms. You shall
+ not use Cocos Creator software for developing other software or tools that's
+ used for developing games. You are not granted to publish, distribute,
+ sublicense, and/or sell copies of Cocos Creator.
+
+ The software or tools in this License Agreement are licensed, not sold.
+ Xiamen Yaji Software Co., Ltd. reserves all rights not expressly granted to you.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ THE SOFTWARE.
+ */
+
+import { EDITOR } from 'internal:constants';
+import { ccclass, editable, executeInEditMode, menu, serializable, type } from 'cc.decorator';
+import { Vec3, Mat4 } from '../core/math';
+import { Node } from '../scene-graph/node';
+import { Component } from '../scene-graph/component';
+import { Camera, CameraProjection } from '../render-scene/scene';
+import { Mesh, MeshRenderer } from '../3d';
+import { assertIsTrue } from '../core/data/utils/asserts';
+import { scene } from '../render-scene';
+import { NodeEventType } from '../scene-graph/node-event';
+
+const _DEFAULT_SCREEN_OCCUPATION: number[] = [0.5, 0.25, 0.125];
+@ccclass('cc.LOD')
+export class LOD {
+    // The relative minimum transition height in screen space.
+    @serializable
+    protected _screenRelativeTransitionHeight = 1;
+    // Mesh renderers components contained in this LOD level.
+    @type([MeshRenderer])
+    @serializable
+    protected _renderers: (MeshRenderer | null)[] = [];
+    // renderer internal LOD data block.
+    protected _LOD: scene.LOD = new scene.LOD();
+
+    constructor () {
+        this._LOD.screenRelativeTransitionHeight = this._screenRelativeTransitionHeight;
+    }
+
+    /**
+     * @en The relvative (minimum) transition height of this LOD level in screen space
+     * @zh 本层级（最小）相对屏幕区域的过渡高度
+     */
+    @type(Number)
+    get screenRelativeTransitionHeight () { return this._screenRelativeTransitionHeight; }
+    set screenRelativeTransitionHeight (val) {
+        this._screenRelativeTransitionHeight = val;
+        this._LOD.screenRelativeTransitionHeight = val;
+    }
+
+    @type([MeshRenderer])
+    get renderers () {
+        return this._renderers;
+    }
+
+    /**
+     * engineInternal, for editor only
+     */
+    set renderers (meshList) {
```

Comment:
The function name is set renderers, but the parameter name is meshList, it is wired. And please add type for parameter.